File: src/ShiftRight.java
//Link: https://codeforces.com/group/MWSDmqGsZm/contest/223205/problem/K

import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class ShiftRight
{
    public static void main(String[]args)
    {
        Scanner cs=new Scanner(System.in);
        int n=cs.nextInt();
        int x=cs.nextInt();
        ArrayList<Integer>v=new ArrayList<>();
        for(int i=0;i<n;i++)
        {
            v.add(cs.nextInt());
        }
        shiftedRight(x,v);
    }

    private static void shiftedRight(int x,ArrayList<Integer>v)
    {
        Collections.rotate(v,x);
        for (int i=0;i<v.size();i++)
        {
            if(i==v.size()-1)
                System.out.print(v.get(i));
            else
                System.out.print(v.get(i)+" ");

        }
    }
}

File: function/Shift Right.cpp
//Link: http://codeforces.com/group/MWSDmqGsZm/contest/223205/problem/K

#include <iostream>
#include <algorithm>
#include<cmath>
#include <vector>
#include <iomanip>
using namespace std;

void shift_right(vector<int>&v, int x)
{
	
		x %= v.size();
		rotate(v.rbegin(), v.rbegin() + x, v.rend());

		for (int i = 0; i < v.size(); i++)
		{
			if (i == v.size() - 1)
				cout << v[i];
			else
			cout << v[i] << " ";
		}
	
}

int main()
{
	int n, x;
	cin >> n >> x;
	vector<int>v(n);
	for (int i = 0; i < n; i++)
	{
		cin >> v[i];
	}

	shift_right(v, x);
}